### ISSUE TITLE
Create "Hello Module" Integration Test

### ISSUE THREAD

**Issue by astorm** [astorm/pestle#119]
## Create Module Integration Test

We need an integration test (the tests in `tests-integration`) that will
1. Use `generate_module` to generate a module
2. Add a URL endpoint with `generate_route`
3. Adds a view with generate_view
4. Tests that the route is available via HTTP, and that the view renders a full Magento page. 

In addition to being a useful test, this should flush out any problems with the new enviornment.

The test in 

```
tests-integration/FirstTest.php
```

has a basic example of how to run a pestle command from a test framework class.  One optional (but not necessary) additional task here would be test helpers of some sort to run command with parameters, as well as test HTTP results.

Integration tests run locally with

```
vendor/bin/phpunit tests-integration
```

and [travis builds](https://github.com/astorm/pestle/blob/master/.travis.yml) with access to a full installed Magento system (available at the URL `http://magento-2-travis.dev`)

**Comment by brideo** [comment]
This may be a bit of a stupid question, however are these tests only for the Pestle repository or should they be accessible from `pestle.phar`?

**Comment by astorm** [comment]
@brideo Good question!  These tests are not accessible via `pestle.phar`, they're only for the pestle repository.  The `tests-integration` tests are  here to make sure that `pestle.phar` continue to behave the same way while we're adding new features or making changes to the codebase.  i.e., to make sure we don't break command X while working on command Y.

The tests in the `tests` folder are closer to Unit tests. Again, not available from pestle itself, but there to help us when we're writing code, and to make sure that someone doesn't inadvertently change important behavior we're relying on.  

Did that answer your question? Do you have any others?  Just let me know!

**Comment by brideo** [comment]
Yep perfect thanks!

**Comment by astorm** [comment]
@brideo Super sorry about this -- I just realized that the travis build enviornment **doesn't** build a working Magento installation n travis. I'd though that we already had integrated the work I did in this repository (https://github.com/astorm/magento2-travis) with this project.  This means you don't have a working enviornment to run your integration tests against, and some of what I've said in this ticket probably doesn't make a lot of sense.

I'll try to get to integrating the build from https://github.com/astorm/magento2-travis into this project -- but feel free to tinker there yourself if you like.

**Comment by astorm** [comment]
@brideo I sent out an comment late last night that (I think?) contained incorrect information.  That said, I want to make sure you're on the right track and understand what I'm after.  I realize I may not have explained it correctly.

I took a look at 

https://github.com/astorm/pestle/blob/master/tests-integration/GenerateRouteTest.php

and that is a good solid test.  

However -- what we really want here is a test that runs a command, but then **actually uses the Magento system to make sure things were generated correctly**.  Our [travis build script](https://github.com/astorm/pestle/blob/master/.travis.yml) should build a Magento enviornment that's available at the (fake) top level `http://magento-2-travis.dev` URL.  

Looking at the tests you've written so far, I wasn't sure if those were just steps on the way to what I've described above, or if I wasn't clear enough in my explanation of what we're after here. 

Does that make sense?

**Comment by brideo** [comment]
Hi @astorm,

That does make sense yeah, the PR was mainly to make sure you were happy with the direction I was heading.

Do you know of any decent articles to get Travis running locally on Mac OS X? If not just link me to http://lmgtfy.com

**Comment by brideo** [comment]
Sorry I have just seen your comment, I do remember reading it. I can only see it in my emails. I will try to get setup now.

**Comment by astorm** [comment]
@brideo No, I don't know of any good tutorials.  I don't have travis installed locally.  My original plan was to have the integration tests use a PHPUnit configuration variables if it needed to know something about the Magento system (i.e. it's URL).  Then developers could set these locally to point to their development instance, and the travis enviornment would point to the Magento system we build via `.travis.yml`.  Does that make sense?

**Comment by astorm** [comment]
@brideo Just a heads up -- I needed to put aside some of the tests in 

```
modified:   tests-integration/GenerateModuleTest.php
modified:   tests-integration/GenerateRouteTest.php
```

by commenting out the tests.  They were breaking the travis build, but the final phar still works.

**Comment by brideo** [comment]
Ah it looks like I didn't account for the fact the tests would be in vendor, I will fix and PR ASAP. Sorry about that.
